### PR TITLE
Add IntEnum (like the standard library's IntEnum)

### PR DESCRIPTION
**Pull request by akx**

### CODE REVIEW

File: tests/test_form_fields.py [hunk 1]
```diff
@@ -26,4 +26,5 @@ def test_bound_form_with_instance():
 
 def test_choices():
     form = get_form()
-    assert form.base_fields["zero2"].choices == [(0, 'ZERO'), (1, 'ONE')]
+    assert form.base_fields["zero2"].choices == [(0, 'Zero'), (1, 'One')]
+    assert form.base_fields["int_enum"].choices == BLANK_CHOICE_DASH + [(0, 'foo'), (1, 'B')]
```
Comment (suutari-ai) [inline tests/test_form_fields.py]: So, is this backward incompatible change, since the code for zero2 stays the same but still the choices change?

Comment (akx) [inline tests/test_form_fields.py]: No – the code for `zero2` actually _does_ change, albeit in a subtle way :)

`IntEnum` used to be imported directly from the stdlib.

However, now we're using our fancier IntEnum, which as the same `choices` behavior as `enumfields.Enum`, i.e. auto-titling the name for a choice label when no explicit label exists.

Before:

```python
from enum import Enum, IntEnum
```

After:

```python
from enumfields import Enum, IntEnum, EnumField, EnumIntegerField
```

Comment (suutari-ai) [inline tests/test_form_fields.py]: Ah, right. I didn't notice that this import change was actually in **tests**/models.py. Yep. This is all good then.

**Comment by suutari-ai** [comment]
LGTM

**Pull request merged**